### PR TITLE
Update deployment methods docs

### PR DESCRIPTION
**Pull request by tczekajlo**
**Proposed changes**:
- Update documentation related to the installation methods 

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by tczekajlo** [comment]
@TyDunn @melindaloubser1 Here are changes related to the rasa docs, a summary for the changes:

- The rasa helm chart is the only recommended installation method for production.

The changes mostly drop other installation methods and add several links to the Rasa X docs where are the biggest changes.

Please go through it and share your thoughts 🙏🏻 , is there something that should be added, is something missing?

**Comment by tczekajlo** [comment]
> you need to deploy your assistant with the Rasa Open Source Helm Chart, deploy Rasa X with its own Helm Chart, and then forward events, correct?

@TyDunn answering your question and moving the conversations here.

Yes, pretty much it's what has to be done to connect Rasa OSS deployment with Rasa X deployment.

**Review by rgstephens — COMMENTED** [review]
@tczekajlo Take a look at my re-write and consolidation of the first paragraph and let me know what you think. I think it would be great if we can describe use of REI for a Rasa OSS install and recommend that as an alternative to docker-compose. Do you think there is time to write that up?

**Comment by tczekajlo** [comment]
> @tczekajlo Take a look at my re-write and consolidation of the first paragraph and let me know what you think. I think it would be great if we can describe use of REI for a Rasa OSS install and recommend that as an alternative to docker-compose. Do you think there is time to write that up?

@rgstephens Thanks a lot for reviewing 🙏🏻  The first paragraph looks good to me 👍🏻 

I think we got some extra time 😄  I've created a separate page that describes how to deploy Rasa OSS by using REI and I included it in the "Alternatives deployment methods". Additionally, the document/page says that we recommend it as an alternative to docker-compose.

WDYT?

**Comment by TyDunn** [comment]
@tczekajlo @rgstephens  I think we need to make this page more clear and nudge user towards Rasa X more. 

I chatted with @rugging24 about this, and this is what we came up with for a potential high level structure of this page:

(Paragraph about building a minimum viable assistants)
1. Deploy Rasa Open Source helm chart
2. Deploy action server helm chart
3. Create connection with action server
4. Deploy Rasa X helm chart
5. Forward events from Rasa Open Source to Rasa X

Wdyt of this?

File: docs/docs/how-to-deploy.mdx [hunk 1]
```diff
@@ -511,10 +479,6 @@ your chosen container registry.
 How you reference the custom action image will depend on your deployment. Pick the relevant documentation for
```
Comment (TyDunn) [inline docs/docs/how-to-deploy.mdx:479]: The action server docs need to be clear about how you use the action server with the open source helm chart imo

**Comment by tczekajlo** [comment]
> @tczekajlo @rgstephens I think we need to make this page more clear and nudge user towards Rasa X more.
> 
> I chatted with @rugging24 about this, and this is what we came up with for a potential high level structure of this page:
> 
> (Paragraph about building a minimum viable assistants)
> 
> 1. Deploy Rasa Open Source helm chart
> 2. Deploy action server helm chart
> 3. Create connection with action server
> 4. Deploy Rasa X helm chart
> 5. Forward events from Rasa Open Source to Rasa X
> 
> Wdyt of this?

Looks good to me. I think we miss a part related to the action server. I'll add missing parts.

**Comment by tczekajlo** [comment]
I've reorganized a bit the "Deploying Your Assistant" section and I added things that we talked about in the previous comments.

@TyDunn @rgstephens  Can you 👀  if you have a minute and share your thoughts?

File: docs/docs/deploy/deploy-rasa.mdx [hunk 2]
```diff
@@ -0,0 +1,275 @@
+---
+id: deploy-rasa
+sidebar_label: Deploy Rasa Assistant
+title: Deploy Rasa Assistant
+description: How to deploy your Rasa Assistant with Kubernetes/Openshift
```
Comment (TyDunn) [inline docs/docs/deploy/deploy-rasa.mdx:5]:
```suggestion
description: The first of three steps to deploy your Rasa assistant
```

File: docs/docs/deploy/deploy-rasa.mdx [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+abstract: This page explains how to deploy Rasa Assistant by using Helm.
```
Comment (TyDunn) [inline docs/docs/deploy/deploy-rasa.mdx:6]:
```suggestion
abstract: This page explains how to deploy Rasa Open Source using Helm. The first of three steps in deploying your Rasa assistant.
```

**Review by TyDunn — CHANGES_REQUESTED** [review]
@tczekajlo This is starting to really take shape! I focused on structure again. I like how you broke it into sections. I want to make that even more clear and explicit...

*Preparing For Production*

Deploying Your Assistant

Step 1: Deploy Rasa Open Source
  a. Set up Rasa Open Source deployment

Step 2: Deploy Action Server
  a. Set up Action Server deployment
  b. Connect Action Server to Rasa Open Source deployment 

Step 3: Deploy Rasa X
  a. Set up Rasa X deployment
  b. Connect Rasa X to Rasa Open Source deployment
  
 **Can we maybe even add this table of contents (with links) to the Introduction section?**

File: docs/docs/deploy/deploy-rasa-x.mdx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import variables from './../variables.json';
+
+## Deploy Rasa X
+
+Visit the [Installation Guide for Rasa X](https://rasa.com/docs/rasa-x/installation-and-setup/installation-guide) where you can learn about available installation methods and
+learn how to deploy Rasa X by following one of them.
+
+## Connect Rasa Open Source to Rasa X
+
+Visit the [Connect Rasa Open Source deployment](https://rasa.com/docs/rasa-x/installation-and-setup/deploy#2-connect-rasa-open-source-deployment-the-rasa-helm-chart) section to learn how to connect your Rasa Assistant to Rasa X deployment.
```
[the remark this answers is not in the document]
Comment (tczekajlo) [reply docs/docs/deploy/deploy-rasa-x.mdx:19]: Yes, exactly.

File: docs/docs/deploy/deploy-rasa-x.mdx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+description: Next steps that make your Rasa Assistant better.
+abstract: This page gives an introduction to Rasa X / Enterprise and where to find more information on how to deploy Rasa X / Enterprise
+  and connect your Rasa Assistant to Rasa X / Enterprise deployment.
+---
+<!-- this file is version specific, do not use `@site/...` syntax -->
+import variables from './../variables.json';
+
+## Deploy Rasa X
+
+Visit the [Installation Guide for Rasa X](https://rasa.com/docs/rasa-x/installation-and-setup/installation-guide) where you can learn about available installation methods and
```
[the remark this answers is not in the document]
Comment (tczekajlo) [reply docs/docs/deploy/deploy-rasa-x.mdx:14]: It points to the page which describes available installation methods for Rasa X.

**Comment by melindaloubser1** [comment]
One note: We should link to the action server deployment instructions from the Rasa SDK docs

**Comment by tczekajlo** [comment]
> One note: We should link to the action server deployment instructions from the Rasa SDK docs

Sounds good to me, @TyDunn WDYT about moving the section related to the Action sever deployment to the Rasa SDK docs?

**Comment by TyDunn** [comment]
> WDYT about moving the section related to Action Server deployment to the Rasa SDK docs?

Sounds good to me

**Comment by tczekajlo** [comment]
@rgstephens @TyDunn Can you give it another look if you have a minute? 🙏🏻  Is there anything else to improve/change or we are good here?

**Pull request merged**